fuel/converters/svhn.py

```diff
@@ -0,0 +1,298 @@
+                num_examples = sum(1 for m in members if '.png' in m.name)
+                progress_bar_context = progress_bar(
+                    name='{} file'.format(split), maxval=len(members),
+                    prefix='Extracting')
+                with progress_bar_context as bar:
+                    for i, member in enumerate(members):
+                        f.extract(member, path=TMPDIR)
+                        bar.update(i)
+            return num_examples
+
+        examples_per_split = {}
+        for split in splits:
+            examples_per_split[split] = extract_tar(split)
+
+        num_examples = sum(examples_per_split.values())
+        split_intervals = {}
+        split_intervals['train'] = (0, examples_per_split['train'])
+        split_intervals['test'] = (
+            examples_per_split['train'],
+            examples_per_split['train'] + examples_per_split['test'])
+        split_intervals['extra'] = (
+            examples_per_split['train'] + examples_per_split['test'],
+            num_examples)
+        split_dict = OrderedDict([
+            (split, OrderedDict([(s, split_intervals[split])
+                                 for s in sources]))
+            for split in splits])
+        h5file.attrs['split'] = H5PYDataset.create_split_array(split_dict)
+
+        def make_vlen_dataset(source):
+            dtype = h5py.special_dtype(vlen=numpy.dtype(source_dtypes[source]))
+            dataset = h5file.create_dataset(
+                source, (num_examples,), dtype=dtype)
+            dataset.dims[0].label = 'batch'
+            shape_labels = source_shape_labels[source]
+            dataset_shapes = h5file.create_dataset(
+                '{}_shapes'.format(source), (num_examples, len(shape_labels)),
+                dtype='uint16')
+            dataset.dims.create_scale(dataset_shapes, 'shapes')
+            dataset.dims[0].attach_scale(dataset_shapes)
+            dataset_shape_labels = h5file.create_dataset(
+                '{}_shape_labels'.format(source), (len(shape_labels),),
+                dtype='S{}'.format(
+                    numpy.max([len(label) for label in shape_labels])))
+            dataset_shape_labels[...] = [
+                label.encode('utf8') for label in shape_labels]
+            dataset.dims.create_scale(dataset_shape_labels, 'shape_labels')
+            dataset.dims[0].attach_scale(dataset_shape_labels)
+
+        for source in sources:
+            make_vlen_dataset(source)
+
+        def get_boxes_and_labels(split):
+            boxes_and_labels = []
+            num_ex = examples_per_split[split]
+            progress_bar_context = progress_bar(
+                '{} digitStruct'.format(split), num_ex)
+            path = split_structs[split]
+            with h5py.File(path, 'r') as f, progress_bar_context as bar:
+                for image_number in range(num_ex):
+                    bbox_group = f['digitStruct']['bbox'][image_number, 0]
```

Comment:
A `f_ds = f['digitStruct']` would make the rest of this block a bit less verbose.